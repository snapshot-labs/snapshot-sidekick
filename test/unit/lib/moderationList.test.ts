import getModerationList from '../../../src/lib/moderationList';
import { SqlFixtures } from '../../fixtures/moderation';

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const mockDbQueryAsync = jest.fn((query: string, args?: string[]): any[] => {
  return [];
});
jest.mock('../../../src/helpers/mysql', () => ({
  __esModule: true,
  default: { queryAsync: (query: string, args?: string[]) => mockDbQueryAsync(query, args) }
}));

describe('moderationList', () => {
  it.each(['flaggedLinks', 'flaggedIps'])('returns only the %s', async field => {
    mockDbQueryAsync.mockImplementationOnce(() => {
      return SqlFixtures[field];
    });

    const list = await getModerationList([field]);

    expect(list).toMatchSnapshot();
  });

  it('returns multiple list: flaggedLinks and flaggedIps', async () => {
    mockDbQueryAsync.mockImplementationOnce(() => {
      return SqlFixtures.flaggedLinks.concat(SqlFixtures.flaggedIps);
    });

    const list = await getModerationList(['flaggedLinks', 'flaggedIps']);
    expect(list).toMatchSnapshot();
  });

  it('ignores invalid fields, and only returns flaggedIps', async () => {
    mockDbQueryAsync.mockImplementationOnce(() => {
      return SqlFixtures.flaggedIps;
    });

    const list = await getModerationList(['a', 'b', 'flaggedIps']);
    expect(list).toMatchSnapshot();
  });

  it('returns all fields by default', async () => {
    mockDbQueryAsync.mockImplementationOnce(() => {
      return Object.values(SqlFixtures).flat();
    });

    const list = await getModerationList();
    expect(list).toMatchSnapshot();
  });
});
