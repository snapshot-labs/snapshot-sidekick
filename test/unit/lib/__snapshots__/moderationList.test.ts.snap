// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`moderationList ignores invalid fields, and only returns flaggedIps 1`] = `
{
  "flaggedIps": [
    "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0",
    "19e36255972107d42b8cecb77ef5622e842e8a50778a6ed8dd1ce94732daca9e",
  ],
}
`;

exports[`moderationList returns all fields by default 1`] = `
{
  "flaggedAddresses": [
    "0x0001",
    "0x0002",
  ],
  "flaggedIps": [
    "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0",
    "19e36255972107d42b8cecb77ef5622e842e8a50778a6ed8dd1ce94732daca9e",
  ],
  "flaggedLinks": [
    "https://gogle.com",
    "https://facebook.com",
  ],
  "flaggedProposalBodyKeywords": [
    "claim drop",
    "token",
  ],
  "flaggedProposalTitleKeywords": [
    "hello",
    "test string",
  ],
  "verifiedTokens": {
    "test": {
      "value": "a",
    },
  },
}
`;

exports[`moderationList returns multiple list: flaggedLinks and flaggedIps 1`] = `
{
  "flaggedIps": [
    "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0",
    "19e36255972107d42b8cecb77ef5622e842e8a50778a6ed8dd1ce94732daca9e",
  ],
  "flaggedLinks": [
    "https://gogle.com",
    "https://facebook.com",
  ],
}
`;

exports[`moderationList returns only the flaggedIps 1`] = `
{
  "flaggedIps": [
    "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0",
    "19e36255972107d42b8cecb77ef5622e842e8a50778a6ed8dd1ce94732daca9e",
  ],
}
`;

exports[`moderationList returns only the flaggedLinks 1`] = `
{
  "flaggedLinks": [
    "https://gogle.com",
    "https://facebook.com",
  ],
}
`;
